language/tools/move-package/src/resolution/digest.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use sha2::{Digest, Sha256};
+use std::path::PathBuf;
+
+use crate::source_package::parsed_manifest::PackageDigest;
+
+pub fn compute_digest(paths: &[PathBuf]) -> Result<PackageDigest> {
+    let mut hasher = Sha256::new();
+
+    for path in paths {
+        if path.is_file() {
+            let contents = std::fs::read(path)?;
+            hasher.update(contents);
+        } else {
+            for entry in walkdir::WalkDir::new(path)
```

Comment:
Looks like `walkdir` doesn't follow symbolic links by default (neither did `checksumdir`). I'm curious, have you given thought to what the behavior ought to be when a symbolic link exists in a source directory? (Sorry if you wrote about this topic in a previous PR!)

Reply:
Oooh very good question. I didn't realize this wasn't done by default. I think we want this to read symlinks since if I understand the semantics of `std::fs::read` it will read the symlink if it's in the source dir, so will add that :)